src/hook/useAppState.ts

```diff
@@ -0,0 +1,28 @@
+import {useRef, useEffect, useState} from 'react';
+import {AppState} from 'react-native';
+
+export function useAppState() {
+  const appStateRef = useRef(AppState.currentState);
+  const [appState, setAppState] = useState({
+    isActive: false,
+    didAppCameToForeground: false,
+  });
+
+  useEffect(() => {
+    const subscription = AppState.addEventListener('change', (nextAppState) => {
+      const isNextStateActive = nextAppState === 'active';
+      setAppState({...appState, isActive: true});
+      if (appStateRef.current.match(/inactive|background/) && isNextStateActive) {
+        setAppState({...appState, didAppCameToForeground: true});
+      }
+      appStateRef.current = nextAppState;
+    });
+
+    return () => {
+      subscription.remove();
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Why not including the dependencies that are used inside the effect?

Reply:
Here the only dependency is `appState` and once the listener is registered this hook does not need to run anymore.

Reply:
It is not needed anymore. The hook does not have any dependency now.